kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -325,7 +325,11 @@ public void initiateConnect(HostPort remote, List<FilterAndInvoker> filters) {
             else {
                 state = State.FAILED;
                 // Close the connection if the connection attempt has failed.
-                LOGGER.trace("Outbound connect error, closing inbound channel", future.cause());
+                Throwable failureCause = future.cause();
+                LOGGER.atWarn()
+                        .setCause(LOGGER.isDebugEnabled() ? failureCause : null)
+                        .log("Connection to target cluster on {} failed, closing inbound channel: {}. Increase log level to DEBUG for stacktrace", remote,
```

Comment:
The upstream failure reasons seem to just appear randomly how about?

```suggestion
                        .log("Connection to target cluster on {} failed with: {}, closing inbound channel. Increase log level to DEBUG for stacktrace", remote,
```